feat(dataplane): unify abstract base class

archerzz opened the pull request:
Right now, dataplane does not default to abstract for base classes of polymorphic sets. We'll apply the same logic for mgmt plane to data plane in https://github.com/Azure/autorest.csharp/pull/2878

For backward compatibility, we use `suppress-abstract-base-class` directive to keep the existing behavior for officially released data plane SDKs.

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

archerzz committed:
feat(dataplane): unify abstract base class

Right now dataplane does not default to abstract for base classes of polymorphic sets. We've apply the same logic for mgmt plane to data plane in https://github.com/Azure/autorest.csharp/pull/2878
For backward compatiblity, we use `suppress-abstract-base-class` directive to keep the existing behavior for officially released data plane SDKs.

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.IoT.TimeSeriesInsights](https://apiview.dev/Assemblies/Review/1b00147a2c2d4123b37388ebdff734be)

heaths reviewed (approved):
Text Analytics and Video Analyzer LGTM

ShivangiReja reviewed (approved):
Search LGTM

archerzz committed:
Merge branch 'main' into dataplane/unify-abstract-base-class

archerzz committed:
bump autorest.csharp

archerzz committed:
update api intereface

archerzz committed:
roll back changes on `Azure.Analytics.Synapse.Artifacts`

Dataflow is used in test caess, though it should be a base type (e.g. abstract)

archerzz committed:
update api interface
